### ISSUE TITLE
False Accessibility - Color Contrast test when using white offscreen text

### ISSUE THREAD

**Issue by Snapstromegon** [dequelabs/axe-core#500]
I filed an issue for the Lighthouse-Team, but they referred to you.
Original Lighthouse Issue:
https://github.com/GoogleChrome/lighthouse/issues/3143#issuecomment-325255609

Here's my post to the Lighthouse-Issues:

I'm currently working on a wordclock and while testing it, I found out, that the Accessibility test marks some of my texts as not readable, but they are white (#ffffff) on black background (#000000).

During the test exactly those elements are getting offscreen (because it stays in the mobile view and my clock makes a square using vh) and get marked as not readable.

If this isn't a problem and I made a mistake I would like to hear what I've done wrong.

**Comment by dylanb** [comment]
Can you create a code snippet or a codepen that shows the problem?

**Comment by Snapstromegon** [comment]
@dylanb Here ist a codepen, but you might need to fiddle around with the view to repoduce the error.
https://codepen.io/Snapstromegon/pen/VzErNy

**Comment by marcysutton** [comment]
A note about testing this, I was able to reproduce it in aXe by using the Responsive Device viewer in Chrome set to a phone size. The items it flags are indeed offscreen–if you try highlighting them or inspecting them, they aren't visible. We could take a closer look at this when we revise color contrast for offscreen techniques, but it might be tricky since it's dynamically calculated.

**Comment by dylanb** [comment]
These are off screen but they are off to the right of the content, so the current algorithm treats it as though it could be scrolled into view. The calculation is correct under these scenarios. I would like to see a proper use case of actual off screen that is not being handled correctly before suggesting that we invest time into this.

**Comment by marcysutton** [comment]
The elements that get pulled up as incomplete are technically off-screen–move the devtools so they don't cover those elements, and aXe can tell precisely what the colors are. If you turn off the `width: 100vw; height: 100vh;` on the body, it passes.

I narrowed it to this part of the code that checks the background stack:

```
if (rect.left > window.innerWidth) {
    return;
}
if (rect.top > window.innerHeight) {
    return;
}
```
Each incomplete result bails out in one of those conditions due to it being offscreen, and the background stack ends up empty. It then either has an equal ratio of white on white because the body background got kicked out of the stack, or it says the body element is overlapping the gray text. I'm not sure what to do about this one, other than say that you have to make the entire page visible to use aXe on it.

**Comment by WilcoFiers** [comment]
This seems to have been fixed half a century ago. Closing the issue.
